File: Arrays/Best_Time_to_Buy_and_Sell_Stock/Best_Time_to_Buy_and_Sell_Stock.java
// Best Time to Buy and Sell Stock
// Java solution goes here...

public class Best_Time_to_Buy_and_Sell_Stock{
    public int calculateMaxProfit (int[] prices) {
        int min=Integer.MAX_VALUE;
        int max=0;
        for (int price : prices)
        {
            if (price < min) {
                min = price;
            } else if (price - min > max) {
                max = price - min;
            }
        }
        return max;
    }

    public static void main(String[] args) {
        Best_Time_to_Buy_and_Sell_Stock bs = new Best_Time_to_Buy_and_Sell_Stock();
        int[] prices = {7,1,5,3,6,4};
        int result = bs.calculateMaxProfit(prices);
        System.out.println(result);
    }

}

File: Arrays/Best_Time_to_Buy_and_Sell_Stock/Best_Time_to_Buy_and_Sell_Stock.py
# Best Time to Buy and Sell Stock
# Python solution goes here...
def max_profit(prices) -> int:
    min_price = float('inf')
    max=0
    for i in prices:
        if i <min_price:
            min_price=i
        elif i-min_price >max:
            max=i-min_price
    return max

def main():
    prices = [7,1,5,3,6,4]
    print(max_profit(prices))

if __name__ == "__main__":
    main()